src/main/java/org/example/springbootintro/controller/BookController.java

```diff
@@ -48,7 +59,15 @@ public void deleteById(@PathVariable Long id) {
 
     @PutMapping("/{id}")
     @ResponseStatus(HttpStatus.OK)
-    public BookDto updateById(@PathVariable Long id, @RequestBody CreateBookRequestDto bookDto) {
+    public BookDto updateById(@PathVariable Long id,
+                              @RequestBody @Valid CreateBookRequestDto bookDto
+    ) {
         return bookService.updateById(id, bookDto);
     }
+
+    @GetMapping("/search")
+    @ResponseStatus(HttpStatus.OK)
+    public List<BookDto> search(@Valid BookSearchParametersDto bookSearchParametersDto) {
+        return bookService.search(bookSearchParametersDto);
+    }
```

Comment:
But it is about:
1) Add meaningful validation for existing DTO classes
So I create my own validator for my search request
Maybe I don't understand something, because as for me it's belongs to this task too

Reply:
Tried to do rebase with Vadym